com.unity.render-pipelines.universal/Editor/ShaderGUI/BaseShaderGUI.cs

```diff
@@ -214,20 +254,55 @@ public void ShaderPropertiesGUI(Material material)
         ////////////////////////////////////
         #region DrawingFunctions
 
+        public void DrawShaderGraphProperties(Material material, IEnumerable<MaterialProperty> properties)
+        {
+            if (properties == null)
+                return;
+
+            foreach (var prop in properties)
+            {
+                if ((prop.flags & (MaterialProperty.PropFlags.HideInInspector | MaterialProperty.PropFlags.PerRendererData)) != 0)
+                    continue;
+
+                float h = materialEditor.GetPropertyHeight(prop, prop.displayName);
+                Rect r = EditorGUILayout.GetControlRect(true, h, EditorStyles.layerMaskField);
+
+                materialEditor.ShaderProperty(r, prop, prop.displayName);
+            }
+        }
+
+        public static void DrawFloatToggleProperty(GUIContent styles, MaterialProperty prop)
```

Comment:
Does it need to be public? At worst maybe protected?

Reply:
I'll set it internal.